--- a/lib/boot/pkg/base/Data/Dynamic.hs~	2018-10-28 19:46:54.000000000 +0000
+++ a/lib/boot/pkg/base/Data/Dynamic.hs	2018-10-28 21:30:10.875014157 +0000
@@ -133,12 +133,15 @@
 
 -- (f::(a->b)) `dynApply` (x::a) = (f a)::b
 dynApply :: Dynamic -> Dynamic -> Maybe Dynamic
+dynApply _ _ = error "Data.Dynamic.dynApply not implemented in ghcjs"
+{-
 dynApply (Dynamic (Fun ta tr) f) (Dynamic ta' x)
   | Just HRefl <- ta `eqTypeRep` ta'
   , Just HRefl <- typeRep @Type `eqTypeRep` typeRepKind tr
   = Just (Dynamic tr (f x))
 dynApply _ _
   = Nothing
+-}
 
 dynApp :: Dynamic -> Dynamic -> Dynamic
 dynApp f x = case dynApply f x of
