--- old/utils/makePackages.sh	2018-07-03 06:51:44.105635958 -0700
+++ new/utils/makePackages.sh	2018-07-03 06:53:28.124074766 -0700
@@ -222,7 +222,7 @@
 cd "$GHCSRC"
 if [ ! -f ./configure ]; then
     echo "booting GHC tree"
-    ./boot
+    $SOURCEDIR/boot
 fi
 
 if [ ! -f ./compiler/ghc.cabal ]; then
--- old/utils/pkg-input/haddock-api-ghcjs/src/Haddock.hs~	2018-07-03 18:25:51.476820305 +0000
+++ new/utils/pkg-input/haddock-api-ghcjs/src/Haddock.hs	2018-07-03 18:25:19.837255552 +0000
@@ -49,6 +49,8 @@
 import Data.IORef
 import Data.Map (Map)
 import qualified Data.Map as Map
+import System.Directory (getCurrentDirectory)
+import System.FilePath ((</>))
 import System.IO
 import System.Exit
 
@@ -479,6 +479,7 @@
 
 getHaddockLibDir :: [Flag] -> IO String
 getHaddockLibDir flags =
+  here <- getCurrentDirectory
   case [str | Flag_Lib str <- flags] of
     [] -> do
 #ifdef IN_GHC_TREE
@@ -487,8 +488,8 @@
       -- if data directory does not exist we are probably
       -- invoking from either ./haddock-api or ./
       let res_dirs = [ getDataDir -- provided by Cabal
-                     , pure "resources"
-                     , pure "haddock-api/resources"
+                     , pure (here </> "resources")
+                     , pure (here </> "haddock-api/resources")
                      ]
 
           check get_path = do
