diff --git a/ghcjs.cabal b/ghcjs.cabal
index 2dc4986..9e02d7b 100644
--- a/ghcjs.cabal
+++ b/ghcjs.cabal
@@ -152,7 +152,7 @@ Library
                    split          >= 0.2      && < 0.3,
                    deepseq,
-                   vector         >= 0.10     && < 0.12,
+                   vector         >= 0.10     && < 0.13,
                    data-default   >= 0.5      && < 0.8,
                    array          >= 0.4      && < 0.6,
                    binary         >= 0.7      && < 0.8,
                    text-binary    >= 0.1      && < 0.3,
--- a/src/Compiler/JMacro/Base.hs	2017-10-09 19:37:41.224262761 -0700
+++ b/src/Compiler/JMacro/Base.hs	2017-10-10 11:52:32.066302614 -0700
@@ -294,7 +294,8 @@
 
 -- | Identifiers
 newtype Ident = TxtI Text
- deriving (Show, Data, Typeable, Hashable, Eq, Ord, Binary, Generic, NFData)
+ deriving (Show, Data, Typeable, Hashable, Eq, Ord, Generic, NFData)
+deriving instance Binary text => Binary (TexI text)
 
 expr2stat :: JExpr -> JStat
 expr2stat (ApplExpr x y) = (ApplStat x y)
