--- a/ghcjs.cabal
+++ b/ghcjs.cabal
@@ -141,7 +141,7 @@
                    mtl,
                    ghc-paths,
                    template-haskell,
-                   syb            >= 0.6      && < 0.7,
+                   syb            >= 0.6      && < 0.8,
                    bytestring     >= 0.10     && < 0.11,
                    attoparsec     >= 0.12     && < 0.14,
                    aeson          >= 0.7      && < 0.12,
@@ -154,7 +154,7 @@
                    transformers,
                    split          >= 0.2      && < 0.3,
                    deepseq,
-                   vector         >= 0.10     && < 0.12,
+                   vector         >= 0.10     && < 0.13,
                    data-default   >= 0.7      && < 0.8,
                    array          >= 0.4      && < 0.6,
                    binary         >= 0.7      && < 0.9,
@@ -172,7 +172,7 @@
                    safe                       >= 0.3  && < 0.4,
                    parsec                     >= 3.1  && < 3.2,
                    haskell-src-exts           >= 1.16 && < 1.19,
-                   haskell-src-meta           >= 0.6.0.3  && < 0.8
+                   haskell-src-meta           >= 0.6.0.3  && < 0.9
     exposed: True
     buildable: True
     hs-source-dirs: src lib/ghcjs-prim
@@ -235,7 +235,7 @@
     Hs-Source-Dirs:   src-bin
     Build-Depends:    ghcjs,
                       base       >= 4   && < 5,
-                      directory  >= 1   && < 1.3,
+                      directory  >= 1   && < 1.4,
                       process    >= 1   && < 1.6,
                       filepath,
                       containers,
@@ -276,13 +276,13 @@
                       text                 >= 0.11   && < 1.3,
                       bytestring           >= 0.10   && < 0.11,
                       system-filepath      >= 0.4    && < 0.5,
-                      shelly               >= 1.5    && < 1.7,
+                      shelly               >= 1.5    && < 1.6.8.5,
                       system-fileio        >= 0.3    && < 0.4,
                       optparse-applicative >= 0.11   && < 0.13,
                       tar                  >= 0.5    && < 0.6,
                       HTTP                 >= 4000.2 && < 5000,
                       yaml                 >= 0.8    && < 0.9,
-                      process              >= 1.2    && < 1.5,
+                      process              >= 1.2    && < 1.7,
                       time                 >= 1.4    && < 1.7,
                       unix-compat          >= 0.4    && < 0.5,
                       executable-path      >= 0      && < 0.1
@@ -344,7 +344,7 @@
                       bytestring,
                       deepseq,
                       unordered-containers,
-                      shelly               >= 1.5 &&  < 1.7,
+                      shelly               >= 1.5 &&  < 1.6.8.5,
                       data-default         >= 0.7 &&  < 0.8,
                       yaml                 >= 0.8 &&  < 0.9,
                       optparse-applicative,
--- a/src-bin/Boot.hs	2017-11-11 10:15:50.945621443 -0800
+++ b/src-bin/Boot.hs	2017-11-11 10:16:21.869975253 -0800
@@ -600,7 +600,7 @@
       branch' <- view (beSources . branch)
       sub $ do
         cd repoName
-        git_ ["checkout", branch']
+        git_ ("checkout" : T.words branch')
     cloneGitSrcs d [] = failWith ("could not clone " <> d <> ", no available sources")
     cloneGitSrcs d (x:xs) = git_ ["clone", x] `catchAny_`
       (msgD warn "clone failed, trying next source" >> cloneGitSrcs d xs)
--- old/src-bin/Pkg.hs.orig	2018-02-05 06:37:38.125577519 -0800
+++ new/src-bin/Pkg.hs	2018-02-05 06:49:24.626280284 -0800
@@ -1114,6 +1114,7 @@
        GhcPkg.extraLibraries     = extraLibraries pkg,
        GhcPkg.extraGHCiLibraries = extraGHCiLibraries pkg,
        GhcPkg.libraryDirs        = libraryDirs pkg,
+       GhcPkg.libraryDynDirs     = libraryDynDirs pkg,
        GhcPkg.frameworks         = frameworks pkg,
        GhcPkg.frameworkDirs      = frameworkDirs pkg,
        GhcPkg.ldOptions          = ldOptions pkg,
