--- old/ghcjs/src-bin/Boot.hs	2016-02-16 11:49:07.026054820 -0800
+++ new/ghcjs/src-bin/Boot.hs	2016-02-16 11:49:37.882054632 -0800
@@ -304,7 +304,7 @@
       installFakes
       installStage1
       unless (e ^. beSettings . bsStage1Unbooted) $ do
-        removeFakes
+        -- removeFakes
         unless (e ^. beSettings . bsQuick) installStage2
       when (e ^. beSettings . bsHaddock) buildDocIndex
       liftIO . printBootEnvSummary True =<< ask
@@ -802,7 +802,7 @@
 installStage2 :: B ()
 installStage2 = subTop' "ghcjs-boot" $ do
   msg info "installing Cabal library"
-  removeFakes
+  -- removeFakes
   cabalPkg <- view (beStages . bstCabal)
   preparePackage cabalPkg
   cabalInstall [cabalPkg]
