--- a/src-bin/Boot.hs	2017-11-11 10:15:50.945621443 -0800
+++ b/src-bin/Boot.hs	2017-11-11 10:16:21.869975253 -0800
@@ -600,7 +600,7 @@
       branch' <- view (beSources . branch)
       sub $ do
         cd repoName
-        git_ ["checkout", branch']
+        git_ ("checkout" : T.words branch')
     cloneGitSrcs d [] = failWith ("could not clone " <> d <> ", no available sources")
     cloneGitSrcs d (x:xs) = git_ ["clone", x] `catchAny_`
       (msgD warn "clone failed, trying next source" >> cloneGitSrcs d xs)
