diff --git a/ghcjs.cabal b/ghcjs.cabal
index 2dc4986..9e02d7b 100644
--- a/ghcjs.cabal
+++ b/ghcjs.cabal
@@ -154,7 +154,7 @@
                    transformers,
                    split          >= 0.2      && < 0.3,
                    deepseq,
-                   vector         >= 0.10     && < 0.12,
+                   vector         >= 0.10     && < 0.13,
                    data-default   >= 0.7      && < 0.8,
                    array          >= 0.4      && < 0.6,
                    binary         >= 0.7      && < 0.9,
