--- a/ghcjs.cabal
+++ b/ghcjs.cabal
@@ -154,7 +154,7 @@
                    transformers,
                    split          >= 0.2      && < 0.3,
                    deepseq,
-                   vector         >= 0.10     && < 0.12,
+                   vector         >= 0.10     && < 0.13,
                    data-default   >= 0.7      && < 0.8,
                    array          >= 0.4      && < 0.6,
                    binary         >= 0.7      && < 0.9,
--- a/src-bin/Boot.hs	2017-11-11 10:15:50.945621443 -0800
+++ b/src-bin/Boot.hs	2017-11-11 10:16:21.869975253 -0800
@@ -600,7 +600,7 @@
       branch' <- view (beSources . branch)
       sub $ do
         cd repoName
-        git_ ["checkout", branch']
+        git_ ("checkout" : T.words branch')
     cloneGitSrcs d [] = failWith ("could not clone " <> d <> ", no available sources")
     cloneGitSrcs d (x:xs) = git_ ["clone", x] `catchAny_`
       (msgD warn "clone failed, trying next source" >> cloneGitSrcs d xs)
