--- old/ghcjs.cabal	2018-05-11 06:22:53.171677911 -0700
+++ new/ghcjs.cabal	2018-05-11 06:25:22.761555360 -0700
@@ -196,7 +196,7 @@
                     directory,
                     filepath,
                     ghcjs,
-                    haddock-api-ghcjs >= 2.15
+                    haddock-api >= 2.15
     GHC-Options:    -Wall
 
 Executable hsc2hs-ghcjs
--- old/utils/pkg-input/haddock-api-ghcjs/src/Haddock.hs
+++ new/utils/pkg-input/haddock-api-ghcjs/src/Haddock.hs
@@ -62,7 +62,7 @@ import System.FilePath
 #else
 import qualified GHC.Paths as GhcPaths
 import Paths_haddock_api_ghcjs (getDataDir)
-import System.Directory (doesDirectoryExist)
+import System.Directory (doesDirectoryExist, getCurrentDirectory)
 #endif
 
 import GHC hiding (verbosity)
@@ -438,10 +438,11 @@ getHaddockLibDir flags =
             exists <- doesDirectoryExist p
             pure $ if exists then Just p else Nothing
 
-      dirs <- mapM check res_dirs  
+      dirs <- mapM check res_dirs
+      here <- getCurrentDirectory
       case [p | Just p <- dirs] of
         (p : _) -> return p 
-        _       -> die "Haddock's resource directory does not exist!\n"
+        _       -> die $ "Haddock's resource directory does not exist!\n  " ++ show dirs ++ ", PWD=" ++ show here ++ "\n"
 #endif
     fs -> return (last fs)
 
--- old/utils/makePackages.sh	2018-05-17 15:25:41.624736849 -0700
+++ new/utils/makePackages.sh	2018-05-17 15:28:25.679242564 -0700
@@ -1,6 +1,8 @@
 #!/usr/bin/env bash
 set -euo pipefail
 
+set -x
+
 # pull the code for GHCJS dependencies from the GHC tree in
 #   lib/ghc
 #
@@ -159,7 +161,7 @@
     SRCDIST="$SRCDISTTMP"
   done
   (
-  cd "$PKGDST"
+  rm -rf "$PKGDST"; mkdir -p "$PKGDST"; cd "$PKGDST"
   echo "unpacking sources: $SRCDIST"
   tar --strip 1 -xzf "$SRCDIST"
   )
@@ -222,7 +222,7 @@
 cd "$GHCSRC"
 if [ ! -f ./configure ]; then
     echo "booting GHC tree"
-    ./boot
+    ../utils/boot
 fi
 
 if [ ! -f ./compiler/ghc.cabal ]; then
